src/test/sql/sql_pipeline_test.cpp

```diff
@@ -0,0 +1,470 @@
+#include <memory>
+#include <string>
+#include <utility>
+#include "../base_test.hpp"
+
+#include "SQLParser.h"
+#include "gtest/gtest.h"
+#include "logical_query_plan/join_node.hpp"
+
+#include "operators/abstract_join_operator.hpp"
+#include "operators/print.hpp"
+#include "operators/validate.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "scheduler/job_task.hpp"
+#include "scheduler/node_queue_scheduler.hpp"
+#include "scheduler/topology.hpp"
+#include "sql/sql_pipeline.hpp"
+#include "sql/sql_query_operator.hpp"
+#include "storage/storage_manager.hpp"
+
+namespace opossum {
+
+class SQLPipelineTest : public BaseTest {
+ protected:
+  void SetUp() override {
+    _table_a = load_table("src/test/tables/int_float.tbl", 2);
+    StorageManager::get().add_table("table_a", _table_a);
+
+    _table_b = load_table("src/test/tables/int_float2.tbl", 2);
+    StorageManager::get().add_table("table_b", _table_b);
+
+    _join_result = std::make_shared<Table>();
+    _join_result->add_column("a", DataType::Int);
+    _join_result->add_column("b", DataType::Float);
+    _join_result->add_column("bb", DataType::Float);
+    _join_result->append({12345, 458.7f, 456.7f});
+    _join_result->append({12345, 458.7f, 457.7f});
+  }
+
+  std::shared_ptr<Table> _table_a;
+  std::shared_ptr<Table> _table_b;
+  std::shared_ptr<Table> _join_result;
+
+  const std::string _select_query_a = "SELECT * FROM table_a";
+  const std::string _invalid_sql = "SELECT FROM table_a";
+  const std::string _join_query =
+      "SELECT table_a.a, table_a.b, table_b.b AS bb FROM table_a, table_b WHERE table_a.a = table_b.a AND table_a.a "
+      "> 1000";
+
+  const std::vector<std::string> _column_names{"a", "b"};
+  const std::vector<std::string> _join_column_names{"a", "b", "bb"};
+
+  bool _subtree_is_equal(const std::shared_ptr<AbstractLQPNode>& got,
```

Comment:
This goes for _subtree_is_equal, _find_in_lqp, _find_in_query_plan: they all could be in testing_assert.hpp, they are not tied to this test. 

_subtree_is_equal's name is misleading, it checks for node types, should probably be _subtree_types_equal or something.

Minor: _find_in_lqp and _find_in_query_plan are not find methods in the stl's algorithm sense (they don't return an element) _contained_in_lqp maybe?